cns/restserver/util.go

```diff
@@ -235,29 +236,52 @@ func (service *HTTPRestService) updateIpConfigsStateUntransacted(req cns.CreateN
 		}
 	}
 
-	// Add the newIpConfigs, ignore if ip state is already in the map
-	service.addIPConfigStateUntransacted(req.NetworkContainerid, newIPConfigs)
+	// Add the newIpConfigs for 3 different conditions
+	// 1. If NC version from NMAgent is larger or same as new NC version, add new IP as available.
+	// 2. If NC version from NMAgent is larger or same as existing NC version, set existing pending programming IP as available and add new IP as pending programming.
+	// 3. If NC version from NMAgent is smaller than existing NC version, add new IP as pending programming.
+	var newNCVersion int
+	var existingNCVersion int
+	newNCVersion, _ = strconv.Atoi(req.Version)
+	existingNCVersion, _ = strconv.Atoi(existingNCVersionInString)
+	// Wait for Ashvin's thread update which querying NMAgent to get latest NC version and save it locally.
+	nmAgentNCVersion := getNCVersionFromNMAgent(req.NetworkContainerid)
+	if nmAgentNCVersion >= newNCVersion {
+		service.addIPConfigStateUntransacted(cns.Available, cns.Available, req.NetworkContainerid, newIPConfigs)
+	} else {
+		if nmAgentNCVersion >= existingNCVersion {
+			service.addIPConfigStateUntransacted(cns.Available, cns.PendingProgramming, req.NetworkContainerid, newIPConfigs)
+		} else {
+			service.addIPConfigStateUntransacted(cns.PendingProgramming, cns.PendingProgramming, req.NetworkContainerid, newIPConfigs)
```

Comment:
Why we need to move existing IPs back to PendingState if they are already available?

Reply:
1. We are NOT make an outboud call. getNCVersionFromNMAgent didn't have implementation yet. It could be the function which fetch NMAgent version locally from what background thread got and saved.
2. We not not moving existing IPs back to Pending State.